azerothcore/azerothcore-wotlk
# fix(DB/Pets): Added `casterGuid` as a primary key to `pet_aura` table.

Fixed #2589.




## Changes Proposed:
-  Added `casterGuid` as a primary key to `pet_aura` table. When pet has two auras with the same id and effects, but from different casters (e.g. periodic damage auras), then the error occurs, because database wants only one aura with the same id and effect mask. Now caster guid is also checked.
-  


## Issues Addressed:
- Closes #2589.



## Tests Performed:

- No tested ingame.


## How to Test the Changes:

1. Create hunter
2. Create pet via command `.pet create`, targeting any beast
3. `.level 18`
4. `.go creature 16319`
5. Send petto attack two different mobs and let them to cast their poison on the pet (pet should have two poisons on self)
6. Kill mobs by `.die`
7. Relog - errors are gone.


## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
